src/extension.ts

```diff
@@ -173,6 +175,12 @@ export function activate(context: vscode.ExtensionContext): void {
                 node = target;
             }
 
+            const deploySubpath: string = workspace.getConfiguration(extensionPrefix).get(configurationSettings.deploySubpath);
+
+            if (deploySubpath) {
+                fsPath = path.join(fsPath, deploySubpath);
```

Comment:
So this won't quite work. We have several entry points for deploying and one of them is right clicking on a folder to deploy. If you right click on the 'bin/Debug/netcoreapp2.0/publish' folder, then you don't want to append 'bin/Debug/netcoreapp2.0/publish' _again_.

I also prefer to make settings as obvious as possible. That's why in the functions extension you see your sub path when you're selecting a workspace, rather than "secretly" appending the path _after_ the prompt:
![screen shot 2018-04-11 at 11 57 36 am](https://user-images.githubusercontent.com/11282622/38637300-9678c5de-3d7f-11e8-8260-f904eba27aec.png)
(I have two workspace folders open '3' and '4', and '4' has deploySubPath set to 'example/path')

Reply:
You can honestly copy over most of the code in [this file](https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/utils/workspace.ts) and replace the code in `util.showWorkspaceFoldersQuickPick`

The only thing that we would lose is some telemetry stuff, but I can add that back after the fact if you're not familiar with our telemetry stuff